FS.common.js

```diff
@@ -501,19 +501,19 @@ var RNFS = {
 
     if (options.begin) {
       subscriptions.push(RNFS_NativeEventEmitter.addListener('DownloadBegin', (res) => {
-        if (res.jobId === jobId) options.begin(res);
+        if (res.jobId === jobId && options.begin) options.begin(res);
       }));
     }
 
     if (options.progress) {
       subscriptions.push(RNFS_NativeEventEmitter.addListener('DownloadProgress', (res) => {
-        if (res.jobId === jobId) options.progress(res);
+        if (res.jobId === jobId && options.progress) options.progress(res);
       }));
     }
 
     if (options.resumable) {
-      subscriptions.push(RNFS_NativeEventEmitter.addListener('DownloadResumable', (res) => {
-        if (res.jobId === joibId) options.resumable(res);
+      subscriptions.push(RNFS_NativeEventEmitter.addListener('DownloadResumable', () => {
+        if (options.resumable) options.resumable();
```

Comment:
Just noticed this PR after submitting #852 to only call resumable for the correct jobId, maybe they should be combined? I'm not familiar with flow so I didn't notice or address that concern...

Reply:
Just wondering if this type of approach would satisfy flow with fewer checks?
```javascript
    var progress = options.progress;

    if (progress) {
      subscriptions.push(RNFS_NativeEventEmitter.addListener('DownloadProgress', (res) => {
        if (res.jobId === jobId) progress(res);
      }));
    }
```